osf/migrations/0177_update_schema_data.py

```diff
@@ -0,0 +1,285 @@
+        else:
+            for keyword in help_text_keywords:
+                if keyword in description.lower():
+                    help = description
+                    description = ''
+                    break
+    else:
+        # if no title, description text is moved to title.
+        title = description
+        description = ''
+
+    return title, description, help, example
+
+def format_property(question, property, index):
+    """
+    Modify a subquestion's qid, title, description, and help text, because
+    these are often absent.
+    - Modify a subquestion's qid to be of the format "parent-id.current-id", to
+      reflect its nested nature, to ensure uniqueness
+    - For the first nested subquestion, transfer the parent's title, description, and help.
+    """
+    property['qid'] = '{}.{}'.format(get_answer_id(question), property.get('id', ''))
+    if not index:
+        title = question.get('title', '')
+        description = question.get('description', '')
+        help = question.get('help', '')
+        if not property.get('title', '') and not property.get('description'):
+            property['title'] = title
+        if not property.get('description', ''):
+            property['description'] = description
+        if not property.get('help', ''):
+            property['help'] = help
+    return property
+
+
+def format_question(state, rs, question, sub=False):
+    """
+    Split the original question from the schema into multiple schema blocks, all of
+    which have the same chunk_id, to link them.
+    """
+    # If there are subquestions, recurse and format subquestions
+    if question.get('properties'):
+        # Creates section or subsection
+        create_block(
+            state,
+            rs.id,
+            block_type='subsection-heading' if sub else 'section-heading',
+            display_text=question.get('title', '') or question.get('description', ''),
+        )
+        for index, property in enumerate(question.get('properties')):
+            format_question(state, rs, format_property(question, property, index), sub=True)
+    else:
+        # All form blocks related to a particular question share the same chunk_id.
+        chunk_id = generate_object_id()
+        title, description, help, example = find_title_description_help_example(rs, question)
+
+        # Creates question title block
+        create_block(
+            state,
+            rs.id,
+            block_type='question-title',
+            display_text=title,
+            help_text='' if description else help,
+            example_text=example,
+            chunk_id=chunk_id
+        )
+
+        # Creates paragraph block (question description)
+        if description:
+            create_block(
+                state,
+                rs.id,
+                block_type='paragraph',
+                display_text=description,
+                help_text=help,
+                chunk_id=chunk_id,
+            )
```

Comment:
Generally trying to map `description` to a `paragraph` block, when the text is suitable for display to everyone.  Otherwise, `description` text that is more `help-text` like is mapped to `help_text` block instead.  There's some subjectivity here -

Example where `description` is mapped to paragraph form-block: [RIDIE Registration - Study Complete]
```json
{
          "qid": "q1",
          "nav": "StudyID",
          "type": "string",
          "format": "text",
          "title": "StudyID",
          "description": "ID assigned to original registration on RIDIE site",
          "help": "",
          "required": true
        }
```

Example where `description` just becomes help text on a question-title block: [Registered Report Protocol Preregistration]
```json
{
        "id": "page3",
        "title": "Manuscript",
        "questions": [{
            "qid": "q6",
            "title": "Attach manuscript",
            "nav": "Manuscript",
            "type": "osf-upload",
            "format": "osf-upload-open",
            "description":"Please attach the manuscript that has been accepted for publication as a Registered Report.",
            "required":true
        }]
    }
```